### PR TITLE
Python node demos

### PR DESCRIPTION
**Pull request by michalkulakowski**

### CODE REVIEW

File: demos/python_demos/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+--extra-index-url https://download.pytorch.org/whl/cpu 
+git+https://github.com/mzegla/optimum-intel.git@infer_request_pool
```
Comment (dtrawins) [inline demos/python_demos/requirements.txt:2]: Add comment why the fork is used

File: demos/python_demos/seq2seq_translation/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+Prerequisites:
+-  image of OVMS with Python support and Optimum installed
+
+Mount the `./servable` which contains:
+- `model.py` and `config.py` - python scripts which are required for execution and uses optimum-intel framework
+- `config.json` - which defines which servables should be loaded
+- `graph.pbtxt` - which defines MediaPipe graph containing python calculator
+
+```bash
+docker run -it --rm -p 11339:11339 -v $PWD/servable:/workspace -e http_proxy=$http_proxy -e https_proxy=$https_proxy -e no_proxy=$no_proxy openvino/model_server:py --config_path /workspace/config.json --log_level DEBUG --port 11339
```
Comment (dtrawins) [inline demos/python_demos/seq2seq_translation/README.md:11]: use a common port 9000

File: demos/python_demos/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+## Prepare OVMS image with Python support and Optimum package installed
+
+From the root of the repository run:
```
Comment (dtrawins) [inline demos/python_demos/README.md:3]: This step is common for all the 3 demos but it might be missed if someone started reading directly form a demo
It might be safer to repeat this step in each demo.
On this page could just just a very short intro and a list of demos.

File: Makefile [hunk 4]
```diff
@@ -138,10 +138,10 @@ ifeq ($(findstring ubuntu,$(BASE_OS)),ubuntu)
   endif
   ifeq ($(BASE_OS_TAG),20.04)
 	INSTALL_DRIVER_VERSION ?= "22.43.24595"
-	DLDT_PACKAGE_URL ?= http://s3.toolbox.iotg.sclab.intel.com/ov-packages/l_openvino_toolkit_ubuntu20_2023.2.0.13089.cfd42bd2cb0_x86_64.tgz
+	DLDT_PACKAGE_URL ?= https://storage.openvinotoolkit.org/repositories/openvino/packages/2023.2/linux/l_openvino_toolkit_ubuntu20_2023.2.0.13089.cfd42bd2cb0_x86_64.tgz
```
Comment (dkalinowski) [inline Makefile:141]: @dtrawins do we want to use public package on main branch? shouldnt we update OV to latest internal version?

File: demos/python_demos/README.md (hunk 3 above)
Comment (mzegla) [reply demos/python_demos/README.md:3]: I suppose we would also need to add git clone of the repo to each demo, so they can get auto tested.

File: demos/python_demos/llm_python_backend/README.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+python3 download_model.py
+```
+
+The model will appear in `./model` directory.
+
+## Run Model Server
+
+Mount the `./model` directory with the model.  
+Mount the `./servable` which contains:
+- `model.py` and `config.py` - python scripts which are required for execution and uses optimum-intel framework
```
[the remark this answers is not in the document]
Comment (mzegla) [reply demos/python_demos/llm_python_backend/README.md:17]: Hugging Face* 
and I'd say utilities instead of framework (because what we mean here is transfromers, distutils etc.). HF feels like something bigger than framework.

File: demos/python_demos/llm_python_backend/client.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import tritonclient.grpc as grpcclient
+
+channel_args = [
+    # Do not drop the connection for long workloads
+    ("grpc.http2.max_pings_without_data", 0),
+]
+client = grpcclient.InferenceServerClient("localhost:11339", channel_args=channel_args)
```
Comment (mzegla) [inline demos/python_demos/llm_python_backend/client.py:22]: If we change ports don't forget this part

File: demos/python_demos/seq2seq_translation/client.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import tritonclient.grpc as grpcclient
+
+client = grpcclient.InferenceServerClient("localhost:11339")
```
Comment (mzegla) [inline demos/python_demos/seq2seq_translation/client.py:18]: If we change ports don't forget this part

File: demos/python_demos/stable_diffusion/README.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+## Run Model Server
+
+Mount the `./model` directory with the model.  
+Mount the `./servable` which contains:
+- `model.py` and `config.py` - python scripts which are required for execution and uses optimum-intel framework
+- `config.json` - which defines which servables should be loaded
+- `graph.pbtxt` - which defines MediaPipe graph containing python calculator
+
+```bash
+docker run -it --rm -p 11339:11339 -v $(pwd)/servable:/workspace:rw -v $(pwd)/model:/model/:rw -e http_proxy=$http_proxy -e https_proxy=$https_proxy -e no_proxy=$no_proxy <IMAGE> --config_path /workspace/config.json --log_level DEBUG --port 11339
```
Comment (mzegla) [inline demos/python_demos/stable_diffusion/README.md:22]: If we change ports don't forget this part

File: demos/python_demos/stable_diffusion/client.py [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import tritonclient.grpc as grpcclient
+import threading
+import time
+from PIL import Image
+from io import BytesIO
+
+client = grpcclient.InferenceServerClient("localhost:11339")
```
Comment (mzegla) [inline demos/python_demos/stable_diffusion/client.py:22]: If we change ports don't forget this part

File: demos/python_demos/llm_python_backend/README.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+- `config.json` - which defines which servables should be loaded
+- `graph.pbtxt` - which defines MediaPipe graph containing python calculator
+
+```bash
+docker run -it --rm -p 9000:9000 -v $PWD/servable:/workspace -v $PWD/model:/model -e http_proxy=$http_proxy -e https_proxy=$https_proxy -e no_proxy=$no_proxy openvino/model_server:py --config_path /workspace/config.json --log_level DEBUG --port 9000
```
Comment (ngrozae) [inline demos/python_demos/llm_python_backend/README.md:32]: 1. are proxy envs needed? we usually don't use it in examples
2. is log_level DEBUG required?
```suggestion
docker run -it --rm -p 9000:9000 -v $PWD/servable:/workspace -v $PWD/model:/model openvino/model_server:py --config_path /workspace/config.json --port 9000
```

**Review by dtrawins — APPROVED** [review]

File: demos/python_demos/llm_python_backend/README.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
+# Llama demo with python node {#ovms_demo_python_llama}
```
[the remark this answers is not in the document]
Comment (dkalinowski) [reply demos/python_demos/llm_python_backend/README.md:1]: This is not really a chatbot since it does not suppor multi question-answer mode, just a single question.

File: Dockerfile.ubuntu [hunk 12]
```diff
@@ -146,8 +147,10 @@ RUN if [[ "$sentencepiece" == "1" || "$NVIDIA" == "1" ]] ; then true ; else exit
...
 WORKDIR /openvino/build
-RUN if [ "$ov_use_binary" == "0" ] ; then true ; else exit 0 ; fi ; cmake -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE -DENABLE_SAMPLES=0 -DNGRAPH_USE_CXX_ABI=1 -DCMAKE_CXX_FLAGS=" -D_GLIBCXX_USE_CXX11_ABI=1 -Wno-error=parentheses "  ..
+RUN if [ "$ov_use_binary" == "0" ]; then true ; else exit 0 ; fi ; if ! [[ $debug_bazel_flags == *"PYTHON_DISABLE=1"* ]]; then true ; else exit 0 ; fi ; cmake -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE -DENABLE_PYTHON=ON -DENABLE_SAMPLES=0 -DNGRAPH_USE_CXX_ABI=1 -DCMAKE_CXX_FLAGS=" -D_GLIBCXX_USE_CXX11_ABI=1 -Wno-error=parentheses "  ..
+RUN if [ "$ov_use_binary" == "0" ] && [[ $debug_bazel_flags == *"PYTHON_DISABLE=1"* ]]; then true ; else exit 0 ; fi ; cmake -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE -DENABLE_SAMPLES=0 -DNGRAPH_USE_CXX_ABI=1 -DCMAKE_CXX_FLAGS=" -D_GLIBCXX_USE_CXX11_ABI=1 -Wno-error=parentheses "  ..
```
Comment (mzegla) [inline Dockerfile.ubuntu:153]: I wonder if maybe we could move that to separate bash script like `install_openvino_from_source.sh` and then call it here with params etc. 
We could do that in a separate PR.

File: Dockerfile.ubuntu [hunk 13]
```diff
@@ -98,6 +98,7 @@ ARG BASE_IMAGE
 ENV DEBIAN_FRONTEND=noninteractive
```
Comment (mzegla) [inline Dockerfile.ubuntu:98]: We will need the same for Redhat. Not necessarily in that PR.

File: demos/python_demos/llm_python_backend/README.md [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+
+## Build image
+
+From the root of the repository run:
+
+```bash
+git clone https://github.com/openvinotoolkit/model_server.git
+cd model_server
+make python_image MEDIAPIPE_DISABLE=0 PYTHON_DISABLE=0
```
Comment (mzegla) [inline demos/python_demos/llm_python_backend/README.md:10]: Can we remove those parameters? It should be obvious. There's no way to build that image without MP and Python.

File: demos/python_demos/llm_python_backend/requirements.txt [hunk 15]
```diff
@@ -0,0 +1,4 @@
+git+https://github.com/huggingface/optimum-intel.git
```
[the remark this answers is not in the document]
Comment (mzegla) [reply demos/python_demos/llm_python_backend/requirements.txt:1]: For running with multiple clients we could use a forked version.

File: demos/python_demos/seq2seq_translation/README.md [hunk 16]
```diff
@@ -0,0 +1,46 @@
+# Seq2seq demo with python node {#ovms_demo_python_seq2seq}
+
+## Build image
+
+From the root of the repository run:
+
+```bash
+git clone https://github.com/openvinotoolkit/model_server.git
+cd model_server
+make python_image MEDIAPIPE_DISABLE=0 PYTHON_DISABLE=0
```
Comment (mzegla) [inline demos/python_demos/seq2seq_translation/README.md:10]: Same as before. `make python_image` should be enough

File: demos/python_demos/stable_diffusion/README.md [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+
+
+## Build image
+
+From the root of the repository run:
+
+```bash
+git clone https://github.com/openvinotoolkit/model_server.git
+cd model_server
+make python_image MEDIAPIPE_DISABLE=0 PYTHON_DISABLE=0
```
Comment (mzegla) [inline demos/python_demos/stable_diffusion/README.md:11]: Same as before - `make python_image` should be enough

File: demos/python_demos/stable_diffusion/requirements.txt [hunk 18]
```diff
@@ -0,0 +1,5 @@
+git+https://github.com/huggingface/optimum-intel.git
```
Comment (mzegla) [inline demos/python_demos/stable_diffusion/requirements.txt:1]: Use fork if multi concurrency is needed

File: demos/python_demos/llm_python_backend/requirements.txt (hunk 15 above)
[the remark this answers is not in the document]
Comment (mzegla) [reply demos/python_demos/llm_python_backend/requirements.txt:1]: ^^

File: demos/python_demos/stable_diffusion/requirements.txt (hunk 18 above)
Comment (mzegla) [reply demos/python_demos/stable_diffusion/requirements.txt:1]: ^^

**Review by mzegla — APPROVED** [review]
